modal/stub.py

```diff
@@ -69,6 +69,15 @@ def check_sequence(items: typing.Sequence[typing.Any], item_type: typing.Type[ty
         raise InvalidError(error_msg)
 
 
+def check_path_dict(
+    dct: typing.Dict[typing.Union[str, os.PathLike], typing.Any], item_type: typing.Type, error_msg: str
```

Comment:
Looks like this is my bad but for extra credit if you want to fix it: we should probably work with `PurePosixPath` here rather than `PathLike`. We don't want Windows paths :upside_down_face:

Reply:
Good point, changing this in a bunch of places. Changing all of our Pathlike usages in similar places to PurePosixPath now